File: app/src/main/java/com/example/adapter/utility/LoggerUtils.java
package com.example.adapter.utility;

import android.util.Log;

import androidx.annotation.NonNull;

/**
 * @author Nathaniel
 * @date 2018/5/29-10:17
 */
public class LoggerUtils {
    private static final String TAG = LoggerUtils.class.getSimpleName();

    public static void logger(String message) {
        logger(TAG, message);
    }

    public static void logger(@NonNull String tag, String message) {
        logger(tag, Level.ERROR, message);
    }

    private static void logger(@NonNull String tag, @NonNull Level level, String message) {
        if (EmptyUtils.isObjectEmpty(message)) {
            message = "logger message is empty in " + LoggerUtils.class.getSimpleName();
        }
        int length = message.length();
        int started = 0;
        int maxLength = 2000;
        int ending = maxLength;
        int passage = (message.length() + maxLength) / maxLength;
        for (int index = 0; index < passage; index++) {
            if (length > ending) {
                logged(tag, level, message.substring(started, ending));
                started = ending;
                ending = ending + maxLength;
            } else {
                logged(tag, level, message.substring(started, length));
                break;
            }
        }
    }

    private static void logged(@NonNull String tag, @NonNull Level level, @NonNull String message) {
        switch (level) {
            case DEBUG:
                Log.d(tag, message);
                break;
            case WARING:
                Log.w(tag, message);
                break;
            case INFO:
                Log.i(tag, message);
                break;
            case ERROR:
                Log.e(tag, message);
                break;
            case ASSERT:
            case VERBOSE:
            default:
                Log.v(tag, message);
                break;
        }
    }

    public enum Level {
        /**
         * 日志级别
         */
        INFO,
        DEBUG,
        WARING,
        ERROR,
        ASSERT,
        VERBOSE
    }
}

File: adapter/src/main/java/com/nathaniel/adapter/utility/LoggerUtils.kt
package com.nathaniel.adapter.utility

import android.util.Log
import com.nathaniel.adapter.utility.EmptyUtils.isObjectEmpty

/**
 * @author Nathaniel
 * @date 2018/5/29-10:17
 */
object LoggerUtils {
    private val TAG = LoggerUtils::class.java.simpleName
    fun logger(message: String) {
        logger(TAG, message)
    }

    fun logger(tag: String, message: String) {
        logger(tag, Level.ERROR, message)
    }

    private fun logger(tag: String, level: Level, message: String) {
        var message = message
        if (isObjectEmpty(message)) {
            message = "logger message is empty in " + LoggerUtils::class.java.simpleName
        }
        val length = message.length
        var started = 0
        val maxLength = 2000
        var ending = maxLength
        val passage = (message.length + maxLength) / maxLength
        for (index in 0 until passage) {
            if (length > ending) {
                logged(tag, level, message.substring(started, ending))
                started = ending
                ending = ending + maxLength
            } else {
                logged(tag, level, message.substring(started, length))
                break
            }
        }
    }

    private fun logged(tag: String, level: Level, message: String) {
        when (level) {
            Level.DEBUG -> Log.d(tag, message)
            Level.WARING -> Log.w(tag, message)
            Level.INFO -> Log.i(tag, message)
            Level.ERROR -> Log.e(tag, message)
            Level.ASSERT, Level.VERBOSE -> Log.v(tag, message)
            else -> Log.v(tag, message)
        }
    }

    enum class Level {
        /**
         * 日志级别
         */
        INFO, DEBUG, WARING, ERROR, ASSERT, VERBOSE
    }
}